src/Dialect/ONNX/Transforms/InstrumentONNXSignaturePass.cpp

```diff
@@ -105,26 +105,36 @@ class InstrumentONNXSignaturePass
           // When one node is selected, print the details of the tensor.
           builder.create<ONNXPrintSignatureOp>(
               loc, fullNameAttr, detail, operAndRes);
+          return true;
         }
+        return false;
       };
-
-      if (isa<func::FuncDialect>(dialect) || isa<ONNXPrintSignatureOp>(op)) {
+      if (isa<func::FuncDialect>(dialect) ||
+          isa<ONNXPrintSignatureOp, KrnlInstrumentOp>(op)) {
```

Comment:
The code in ONNX/Transforms is actually handling multiple dialects, onnx and krnl. We may think of defining an instrumentation related dialect and/or dialect. Then our work of instrumentation can be widely used.

Reply:
@chentong319:

Right now, the user can specify (with a regex pattern) exactly which operation(s) they want.
With the `-profile-ir` option, we set for ZHigh:
```
    instrumentSignatures = "onnx.*,zhigh.*";
```
and for Onnx, then only the `onnx.*`. 

Is that sufficient for this PR, or would you like to have some additional constraints?